lib/wpc_proto/wpc_proto.c

```diff
@@ -178,14 +181,76 @@ app_proto_res_e WPC_Proto_handle_request(const uint8_t * request_p,
     else if (wp_message_req_p->get_scratchpad_status_req)
     {
         LOGI("Get scratchpad status request\n");
+        resp_size  = sizeof(wp_GetScratchpadStatusResp);
+        resp_msg_p = Platform_malloc(resp_size);
+        if (resp_msg_p == NULL)
+        {
+            LOGE("Not enough memory to encode GetScratchpadStatusResp");
+            return APP_RES_PROTO_NOT_ENOUGH_MEMORY;
+        }
+        message_resp.wirepas->get_scratchpad_status_resp
+            = (wp_GetScratchpadStatusResp *) resp_msg_p;
+
+        res = Proto_config_handle_get_scratchpad_status(wp_message_req_p->get_scratchpad_status_req,
+                                                        (wp_GetScratchpadStatusResp *) resp_msg_p);
     }
     else if (wp_message_req_p->upload_scratchpad_req)
     {
         LOGI("Upload scratchpad request\n");
+        resp_size  = sizeof(wp_UploadScratchpadResp);
+        resp_msg_p = Platform_malloc(resp_size);
+        if (resp_msg_p == NULL)
+        {
+            LOGE("Not enough memory to encode UploadScratchpadResp");
+            return APP_RES_PROTO_NOT_ENOUGH_MEMORY;
+        }
+        wp_UploadScratchpadResp * upload_scratchpad_resp_p = resp_msg_p;
+        message_resp.wirepas->upload_scratchpad_resp = upload_scratchpad_resp_p;
+
+
+        res = Proto_otap_handle_upload_scratchpad(wp_message_req_p->upload_scratchpad_req,
+                                                  upload_scratchpad_resp_p);
+
+        /* Update parameters */
+        Proto_config_initialize_otap_variables();
+
+        if (upload_scratchpad_resp_p->header.res == wp_ErrorCode_OK)
+        {
+            if(wp_message_req_p->upload_scratchpad_req->has_scratchpad)
+            {
+                /* Do some sanity check */
+                app_scratchpad_status_t otap_status = Proto_config_get_otap_status();
```

Comment:
Shouldn't this code belong to Proto_otap_handle_upload_scratchpad?

It seems a bit weird to modify the return code. And it is anyway not possible with upcoming change with upload as chunk, so better to remove it for now.

Reply:
It was moved to wpc_proto.c, then removed completely.